src/applications/personalization/profile/util/index.js

```diff
@@ -36,16 +36,25 @@ export async function getData(apiRoute, options) {
 
 const hasErrorMessage = (errors, errorKey, errorText) => {
   if (errorText) {
-    return errors.some(err =>
-      err.meta?.messages?.some(
-        message =>
-          message.key === errorKey &&
-          message.text?.toLowerCase().includes(errorText.toLowerCase()),
-      ),
+    return (
+      errors.some(err =>
+        err.meta?.messages?.some(
+          message =>
+            message.key === errorKey &&
+            message.text?.toLowerCase().includes(errorText.toLowerCase()),
+        ),
+      ) ||
+      errors.some(
+        err =>
+          err?.code === errorKey &&
+          err?.detail?.toLowerCase().includes(errorText.toLowerCase()),
+      )
```

Comment:
This is new error checking for Lighthouse based response. The first section of error checking (before the || )can be removed once migration to new endpoint is complete.